### PR TITLE
feat(consumer-restriction):  customize rejected_msg

### PR DESCRIPTION
**Pull request by qq54903099**
### What this PR does / why we need it:

consumer-restriction plugin add reject_message parameter support user to custom define return message when request rejected.

### Pre-submission checklist:

* [x] Did you explain what problem does this PR solve? Or what new features have been added?
* [x] Have you added corresponding test cases?
* [x] Have you modified the corresponding document?
* [x] Is this PR backward compatible? **If it is not backward compatible, please discuss on the [mailing list](https://github.com/apache/apisix/tree/master#community) first**

### CODE REVIEW

**Comment by qq54903099** [comment]
how to chose reviewers?

File: apisix/plugins/consumer-restriction.lua [hunk 1]
```diff
@@ -105,7 +106,11 @@ local function is_method_allowed(allowed_methods, method, user)
 end
 
 local function reject(conf)
-    return conf.rejected_code, { message = "The " .. conf.type .. " is forbidden." }
+    local reject_message = "The " .. conf.type .. " is forbidden." ;
```
Comment (spacewander) [inline apisix/plugins/consumer-restriction.lua:109]: We can check `conf.reject_message` before creating new string?

**Review by spacewander — COMMENTED** [review]
Let's check the feature in https://github.com/apache/apisix/blob/139c3972898f1e45c6d5b1f40143c504116b3486/t/plugin/consumer-restriction.t#L246

File: apisix/plugins/consumer-restriction.lua (hunk 1 above)
Comment (qq54903099) [reply apisix/plugins/consumer-restriction.lua]: ok

Comment (qq54903099) [reply apisix/plugins/consumer-restriction.lua]: > 

ok

File: apisix/plugins/consumer-restriction.lua [hunk 2]
```diff
@@ -51,7 +51,8 @@ local schema = {
                 }
             }
         },
-        rejected_code = {type = "integer", minimum = 200, default = 403}
+        rejected_code = {type = "integer", minimum = 200, default = 403},
+        rejected_message = {type = "string"}
```
Comment (leslie-tsang) [inline apisix/plugins/consumer-restriction.lua:55]:
```suggestion
        rejected_msg = {type = "string"}
```
Would it be better to change `rejected_message` to `rejected_msg`? to keep it similar to other plugins.

File: t/plugin/consumer-restriction.t [hunk 3]
```diff
@@ -263,7 +263,8 @@ Authorization: Basic amFjazIwMjA6MTIzNDU2
                             "consumer-restriction": {
                                  "blacklist": [
                                      "jack1"
-                                 ]
+                                 ],
+                                 "rejected_message": "request is forbidden"
```
Comment (leslie-tsang) [inline t/plugin/consumer-restriction.t:267]: New test cases are needed to ensure that the new feat works as expected.

Comment (spacewander) [reply t/plugin/consumer-restriction.t:267]: This change is already a new case. But we need to make sure it can pass.

Comment (spacewander) [reply t/plugin/consumer-restriction.t:267]: @qq54903099 
Please update the test to make CI pass: https://github.com/apache/apisix/runs/4465467353?check_suite_focus=true

File: apisix/plugins/consumer-restriction.lua (hunk 2 above)
Comment (qq54903099) [reply apisix/plugins/consumer-restriction.lua]: ok

**Review by spacewander — APPROVED** [review]

**Review by tokers — APPROVED** [review]

**Pull request merged**